source/AI.cpp

```diff
@@ -141,6 +143,45 @@ namespace {
 				bay.ship->SetCommands(Command::DEPLOY);
 	}
 
+	// Helper function for selecting the ships for formation commands.
+	vector<Ship *> GetShipsForFormationCommand(const PlayerInfo &player)
+	{
+		// Figure out what ships we are giving orders to
+		vector<Ship *> targetShips;
+		auto &selectedShips = player.SelectedShips();
+		bool fullFleet = selectedShips.empty();
+		if(fullFleet)
+		{
+			auto &playerShips = player.Ships();
+			targetShips.reserve(playerShips.size() - 1);
+			for(const shared_ptr<Ship> &it : player.Ships())
+				if(it.get() != player.Flagship() && !it->IsParked()
```

Comment:
```suggestion
				if(it != player.FlagshipPtr() && !it->IsParked()
```

Reply:
using FlagshipPtr is not desirable as it has a lot more logic attached to it and is not just a simple return

Reply:
~~it's totally not like Flagship calls FlagshipPtr~~

Reply:
huh, when did that change, forget what I said

Reply:
Was not working as expected; player is const in this context, and FlagshipPtr itself is not const.

Reply:
> Was not working as expected; player is const in this context, and FlagshipPtr itself is not const in relation to the player class.

Neither is Flagship?

Reply:
It cheats, but externally it is const; `const Ship *PlayerInfo::Flagship() const`

Reply:
oh, I see